src/app/tasks/task-details/task-details.component.ts

```diff
@@ -81,17 +82,23 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
     this.deleteTaskSubscription.unsubscribe();
     this.getCurrentTaskSubscription.unsubscribe();
+    this.updateTaskSubscription.unsubscribe();
   }
   
   ngOnInit(): void {
-    
+    this.getCurrentTask();
+  }
+
+  getCurrentTask() {
     // Get task id from route parameters then pass it as argument for task service
     // set the component task when async method is done
     this.getCurrentTaskSubscription = this.route.params.pipe(
       map(params => params['id'] as number),
       switchMap(taskId => {
         return this.taskService.getTask(taskId)
-      })).subscribe(responseTask => {
+      }),
+      take(1)
```

Comment:
if you do use this approach you don't have to use a Subscription variable to unsubscribe from it later, you can remove this.getCurrentTaskSubscription :)

Reply:
since it will automatically unsubscribe after the 1st value itself (from using this take operator)